packages/common/src/api/tan-query/useRelatedArtists.ts

```diff
@@ -0,0 +1,50 @@
+import { useQuery, useQueryClient } from '@tanstack/react-query'
+import { useDispatch } from 'react-redux'
+
+import { userMetadataFromSDK } from '~/adapters'
+import { transformAndCleanList } from '~/adapters/utils'
+import { useAppContext } from '~/context/appContext'
+import { ID, Id, OptionalId } from '~/models/Identifiers'
+import { MAX_PROFILE_RELATED_ARTISTS } from '~/utils'
+
+import { QUERY_KEYS } from './queryKeys'
+import { useCurrentUserId } from './useCurrentUserId'
+import { primeUserData } from './utils/primeUserData'
+
+type Config = {
+  limit?: number
+  staleTime?: number
+  enabled?: boolean
+}
+
+export const useRelatedArtists = (artistId?: ID | null, config?: Config) => {
+  const { audiusSdk } = useAppContext()
+  const dispatch = useDispatch()
+  const { data: currentUserId } = useCurrentUserId()
+  const queryClient = useQueryClient()
+
+  return useQuery({
+    queryKey: [QUERY_KEYS.relatedArtists, artistId],
+    queryFn: async () => {
+      if (!artistId || !audiusSdk) {
+        return []
+      }
+      const { data } = await audiusSdk.full.users.getRelatedUsers({
+        id: Id.parse(artistId),
+        limit: config?.limit ?? MAX_PROFILE_RELATED_ARTISTS,
+        userId: OptionalId.parse(currentUserId)
+      })
+
+      const users = transformAndCleanList(data, userMetadataFromSDK)
+
+      // Prime the users data in both React Query and Redux
+      if (users.length) {
+        primeUserData({ users, queryClient, dispatch })
```

Comment:
oh nice, did you update this to accept a list now? ty! was on my todo list

Reply:
yeah heads up got all of them working with lists in a subsequent pr so dont worry about it